### DISCUSSION TITLE
Call to a member function chain() on null

### DISCUSSION THREAD

**Discussion by nbyloff** [spatie/laravel-queueable-action#101] [Q&A]
I am unsure on even asking this question because it's probably so simple but cannot get a working example. The code below gives me the error `Call to a member function chain() on null`. The examples don't show any actions returning anything and I cannot find examples anywhere else so I am unsure how to proceed or how to debug.

````
$layout = 'some relevant string data';
$args = [
      'user' => $user,
      'product' => $product
    ];
    app(GeneratePdfAction::class)
      ->execute($layout, $args)
      ->chain([
        new ActionJob(StoreFileAction::class, $args),
      ]);
````

I'd like to use this package to pass data from GeneratePdfAction to StoreFileAction. Is it intended be used in this way? I read somewhere you can pass data from one action to the other within a chain, but I don't see an example of that here either.

**Comment by ejntaylor** [comment]
Chaining seems to work but I get larastan errors with it.

An alternative is to set both as ActionJob classes and use Bus::chain().

It's not ideal and testing is a hog - you have to use ReflectionClass to access the property actionClass for testing..

As you say, hopefully there is something simple we are overlooking but I can't see it atm.

Example:

```
        $firstJob = new ActionJob($this->firstJob::class, [$data]);
        $secondJob = new ActionJob($this->secondJob::class, [$data]);


        Bus::chain([
            $firstJob,
            $secondJob
        ])
            ->onQueue('updates')
            ->dispatch();
```
